### PR TITLE
Adding capability to support PKI

### PR DESCRIPTION
**Pull request by JavaJoeS**
Update with option to install PKI from marketplace, org.eclipse.core.pki

### CODE REVIEW

File: bundles/org.eclipse.equinox.p2.transport.ecf/src/org/eclipse/equinox/internal/p2/transport/ecf/RepositoryTransport.java [hunk 1]
```diff
@@ -95,7 +97,18 @@ public RepositoryTransport(IProvisioningAgent agent) {
...
 		boolean useJREHttp = false;
+		try {
+			// DELMARVA - is org.eclipse.core.pki installed and configured
+			if (SSLContext.getDefault().getProvider().getName().contains("TLS")) { //$NON-NLS-1$
+				useJREHttp = true;
+				Activator.getDefault().useJREHttpClient();
+			}
+		} catch (NoSuchAlgorithmException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
```
Comment (mickaelistria) [inline bundles/org.eclipse.equinox.p2.transport.ecf/src/org/eclipse/equinox/internal/p2/transport/ecf/RepositoryTransport.java:110]: Please use `ILog.log().error(e.getMessage(), e)`

Comment (JavaJoeS) [reply bundles/org.eclipse.equinox.p2.transport.ecf/src/org/eclipse/equinox/internal/p2/transport/ecf/RepositoryTransport.java:110]: @mickaelistria  I noticed that most of the other try catch blocks just throw and dont print anything out.  There are no other ILog messages in that bundle.  Maybe better to stick with the picture show and just throw?

Comment (JavaJoeS) [reply bundles/org.eclipse.equinox.p2.transport.ecf/src/org/eclipse/equinox/internal/p2/transport/ecf/RepositoryTransport.java:110]: @mickaelistria   Looking again, the software is supposed to be transparent and not required, so perhaps btter to not do anything in the catch block.  Software does not even need to know an exception may have occurred.

**Comment by JavaJoeS** [comment]
@mickaelistria Please recheck..

File: bundles/org.eclipse.equinox.p2.transport.ecf/src/org/eclipse/equinox/internal/p2/transport/ecf/RepositoryTransport.java (hunk 1 above)
Comment (merks) [reply bundles/org.eclipse.equinox.p2.transport.ecf/src/org/eclipse/equinox/internal/p2/transport/ecf/RepositoryTransport.java]: Please 🙏 try to keep a PR to a single commit. Amend and force push further changes. Then you won’t be asked to squash the commits later.

**Comment by laeubi** [comment]
> Update with option to install PKI from marketplace, org.eclipse.core.pki

I must confess I can't connect that statement to the actual code, the code seem to enable `useJREHttp`  if the default SSL context contains TLS (what else could it be), how is this connected to `org.eclipse.core.pki`? Why is this doen with P2 transport directly and not with ECF where the actual connection is made?

**Comment by JavaJoeS** [comment]
@laeubi When there is no PKI enabled the default SSLContext is not TLS.

**Review by laeubi — CHANGES_REQUESTED** [review]
Please rebase, squash commits and add documentation somewhere why / what this change does, forcing JRE http does not look "right" just because TLS is enabled (?)

**Comment by laeubi** [comment]
Closing this due to missing feedback from the author, if this is still required please reopen with appropriate explanation and at best some test.

**Pull request closed without merging**